browser/brave_wallet/brave_wallet_tab_helper.cc

```diff
@@ -108,6 +110,36 @@ void BraveWalletTabHelper::BindSolanaProvider(
       std::move(receiver));
 }
 
+// static
+void BraveWalletTabHelper::BindCardanoProvider(
+    content::RenderFrameHost* const frame_host,
+    mojo::PendingReceiver<mojom::CardanoProvider> receiver) {
+  if (!IsCardanoDAppSupportEnabled()) {
+    return;
+  }
+  auto* brave_wallet_service = BraveWalletServiceFactory::GetServiceForContext(
+      frame_host->GetBrowserContext());
+  if (!brave_wallet_service) {
+    return;
+  }
+  auto* host_content_settings_map =
+      HostContentSettingsMapFactory::GetForProfile(
+          frame_host->GetBrowserContext());
+  if (!host_content_settings_map) {
+    return;
+  }
+  content::WebContents* web_contents =
+      content::WebContents::FromRenderFrameHost(frame_host);
+
+  auto* tab_helper = BraveWalletTabHelper::FromWebContents(web_contents);
+  if (!tab_helper) {
+    return;
+  }
+
+  tab_helper->cardano_provider_receivers_.Add(
+      std::make_unique<CardanoProviderImpl>(), std::move(receiver));
```

Comment:
I'd initialize this like for ETH sooner rather than later:
```
      std::make_unique<EthereumProviderImpl>(
          host_content_settings_map, brave_wallet_service,
          std::make_unique<BraveWalletProviderDelegateImpl>(web_contents,
                                                            frame_host),
          prefs),
      std::move(receiver));
```
At least this file would not be touched again.
 
I'm also ok with a followup